app/controllers/api/v1/profiles_controller.rb

```diff
@@ -15,19 +13,26 @@ def update
     @user.update!(profile_params)
   end
 
+  def availability
+    @user.account_users.find_by!(account_id: availability_params[:account_id]).update!(availability: availability_params[:availability])
+  end
+
   private
 
   def set_user
     @user = current_user
   end
 
+  def availability_params
```

Comment:
at the moment configuring `auto_offline` is not available at the agent level. admin determines it.